test/routes/parcelActions.test.js

```diff
@@ -62,3 +62,199 @@ describe('Parcel Actions route test', () => {
+  })
+
+  test('provides correct arguments to rules engine runner', async () => {
+    const options = generateRequestOptions()
+    const sampleParcel = getSampleParcel()
+    const sampleAction = getSampleAction()
+    parcelService.getByRef.mockReturnValue(sampleParcel)
+    actionsService.getByIdWithRules.mockResolvedValue(sampleAction)
+
+    await server.inject(options)
+
+    expect(rulesEngineHelper.fullRun).toHaveBeenCalledWith(
+      expect.objectContaining(sampleAction),
+      expect.objectContaining(sampleParcel),
+      expect.objectContaining({ quantity: sampleAction.input.lowerbound })
+    )
+  })
+
+  test('provides input description as given in action', async () => {
+    const sampleAction = getSampleAction()
+    actionsService.getByIdWithRules.mockResolvedValue(sampleAction)
+    const response = await server.inject(generateRequestOptions())
+    const responseData = JSON.parse(response.payload)
+
+    expect(responseData).toEqual(
+      expect.objectContaining({
+        id: sampleAction.id,
+        description: sampleAction.description,
+        input: expect.objectContaining(sampleAction.input)
+      })
+    )
+  })
+
+  test('elaborates input description with upperbound provided by rules engine runner', async () => {
+    const upperbound = getUpperboundFact(40)
+    rulesEngineHelper.fullRun.mockResolvedValue({ eligible: true, value: 100, upperbound })
+    const response = await server.inject(generateRequestOptions())
+    const responseData = JSON.parse(response.payload)
+
+    expect(responseData.input).toEqual(
+      expect.objectContaining({ upperbound: upperbound.value })
+    )
+  })
+
+  test('rounds upperbound provided by rules engine runner to two decimal places', async () => {
+    const testCases = [
+      { upperbound: getUpperboundFact(739.3000000000001), expectedValue: 739.3 },
+      { upperbound: getUpperboundFact(40.2699999999999), expectedValue: 40.27 },
+      { upperbound: getUpperboundFact(78.8383838383838), expectedValue: 78.84 },
+      { upperbound: getUpperboundFact(44.4444444444444), expectedValue: 44.44 },
+      { upperbound: getUpperboundFact(55.5555555555555), expectedValue: 55.56 }
+    ]
+    for (const testCase of testCases) {
+      const { upperbound, expectedValue } = testCase
+      rulesEngineHelper.fullRun.mockResolvedValue({ eligible: true, value: 100, upperbound })
+      const response = await server.inject(generateRequestOptions())
+      const responseData = JSON.parse(response.payload)
+
+      expect(responseData.input).toEqual(
+        expect.objectContaining({ upperbound: expectedValue })
+      )
+    }
+  })
+
+  test('handles failing rules run', async () => {
+    rulesEngineHelper.fullRun.mockResolvedValue({ eligible: false })
+    expect(async () => {
+      await server.inject(generateRequestOptions())
+    }).not.toThrow()
+  })
+
+  // also need test for action lowerbound being passed to rules runner, rather than 1...
```

Comment:
I don't think this comment is needed. I found the test description below more helpful.

Reply:
Gah, meant to delete it, this was a todo from last night!